data/users.js

```diff
@@ -2,66 +2,71 @@ const { ObjectId } = require('mongodb');
 const mongoCollections = require('../config/mongoCollections');
 const users = mongoCollections.users;
 const products = require('./products');
+const new_user_schema = require('../schemas/new_user');
+const bcrypt = require('bcrypt');
 
 async function get(id) {
+  if (typeof id !== 'string') throw `id must be a string but ${typeof id} provided`;
+
   const usersCollection = await users();
   let user = await usersCollection.findOne({ _id: ObjectId(id) });
 
   if (user !== null) {
     user._id = user._id.toString();
-    try{
-      user.cart = user.cart.map((id) => id.toString());
-    }catch(err) {
-      user.cart = user.cart
-    }
-    
+    user.cart = user.cart.map((id) => id.toString());
   }
 
   return user;
 }
 
-async function create(user) {
-  let newUser = {
-    'email': user.email,
-    'firstName': user.firstName,
-    'lastName': user.lastName,
-    'admin': user.admin,
-    'password': user.password,
-    'address': {
-      'address': user.address.address,
-      'country': user.address.country,
-      'city': user.address.city,
-      'zipcode': user.address.zipcode
-    },
-    'cart': []
-  };
-  
+async function getByEmail(email) {
+  if (typeof email !== 'string') throw `email must be a string but ${typeof email} provided`;
+
   const usersCollection = await users();
-  const insertInfo = await usersCollection.insertOne(newUser);
-  if (insertInfo.insertedCount === 0) throw 'Could not add new user';
+  let user = await usersCollection.findOne({ email: email });
```

Comment:
Question: From the lecture on auth, should this be forced to lower case as it is acting as the username field, or did the professor only want that check done for new user registration? 
```suggestion
  let user = await usersCollection.findOne({ email: email.toLowerCase() });
```

Reply:
Yeah, you're right, I made sure to lowercase the registration email but not the login email. I'll add a few more lowercases to be safe (in data AND routes) and then merge this in.